Первая часть проекта

CzechJD opened the pull request.

CzechJD committed:
Первая часть проекта

CzechJD committed:
Первая часть проекта

CzechJD committed:
1) Поменял пакет для @RequestBody: теперь в поля записываюся передаваемые значения, вместо null
2) Добавил starter-validation в pom.xml для валидации данных
3) Немного изменил ошибки, удалив CapacityException

psorokin02 commented on src/main/java/org/example/location/LocationService.java:
```diff
@@ -0,0 +1,54 @@
+        return mapper.fromEntityToLocation(savedEntity);
+    }
+
+    public Location updateLocation(Integer id, Location location) {
```
по условию нельзя при обновлении локации увеличивать вместимость, нужно это проверять

psorokin02 commented on src/main/java/org/example/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,43 @@
+
+    @ExceptionHandler(RuntimeException.class)
+    protected ResponseEntity<ErrorMessageResponse> handleCapacityException(RuntimeException e) {
+        return buildErrorResponse("Некорректная вместимость локации", e.getMessage(), HttpStatus.BAD_REQUEST);
```
У тебя этот обработчик сработает для всех (RuntimeException.class), а не только тогда, когда вместимость локации неправильная, нужно убрать

psorokin02 commented on src/main/java/org/example/exception/ErrorMessageResponse.java:
```diff
@@ -0,0 +1,16 @@
+@AllArgsConstructor
+public class ErrorMessageResponse {
+
+    private String message;
```
final поля сделать

psorokin02 commented on src/main/java/org/example/location/LocationDto.java:
```diff
@@ -0,0 +1,21 @@
+@Data
+@AllArgsConstructor
+public class LocationDto {
+    private Integer id;
```
final поля сделать

psorokin02 commented on src/main/java/org/example/location/Location.java:
```diff
@@ -0,0 +1,13 @@
+@Data
+@AllArgsConstructor
+public class Location {
+    private Integer id;
```
final поля сделать

psorokin02 commented on src/main/java/org/example/location/LocationService.java:
```diff
@@ -0,0 +1,54 @@
+        LocationEntity savedEntity = repository.save(foundEntity);
+        return mapper.fromEntityToLocation(savedEntity);
+    }
+    public void deleteLocation(Integer id) {
```
форматирование поехало, поправить

CzechJD committed:
Изменения 1.4:
Добавил final
Добавил проверку при обновлении локации
Изменил RuntimeException на IllegalStateException

CzechJD committed:
Изменения 1.5
Добавил final поля
Изменил исключения
